airbyte-webapp/src/utils/connectors/destinations.json

```diff
@@ -0,0 +1,50 @@
+{
+  "AzureBlobStorage": "b4c5d105-31fd-4817-96b6-cb923bfc04cb",
+  "AmazonSqs": "0eeee7fb-518f-4045-bacc-9619e31c43ea",
```

Comment:
It would be nicer if we could pull this mapping directly from the [source_definitions.yaml](https://github.com/airbytehq/airbyte/blob/master/airbyte-config/init/src/main/resources/seed/source_definitions.yaml) and [destination_definitions.yaml](https://github.com/airbytehq/airbyte/blob/master/airbyte-config/init/src/main/resources/seed/destination_definitions.yaml) files, since those are the real source of truth for these IDs, and that way we wouldn't need to maintain these separate JSON files or add to them whenever a new connector is added.

However, I think pulling YAML files in is harder than JSON, and those files are outside the airbyte-webapp which makes them harder to pull in as well. Maybe we can make a symlink, and use some library to convert to JSON?

Reply:
Good call. I'll try to play around. We could potentially symlink as you mentioned, and use a yaml webpack loader to import them.

Reply:
Okay problem here is: if we go this way, we'd lose strong typing on `ConnectorIds.Sources/Destinations`, i.e. because it's not coming from a JSON, we'd either need to manually type it, so it knows which connector names exist on it (which can then also always run async with the actual content of the yaml), or just type it as `{ [key: string]: string }`, which basically means TS won't be able to check anymore if the key you're trying to access actually exists.

I personally would find the loss of the typing here more severe than having this file separate from the specification yamls. We also don't NEED to keep those JSONs constantly up to date, since we mainly need to refresh them in case we actually want to use a (new) connector's id in code, that isn't yet in the JSON. I'd not necessarily try to keep it constantly updated when adding new connectors, if we're not accessing it's ID anywhere.

What do you think?

Reply:
That's a good point - the strong typing here seems useful, and we don't need to add an entry until it actually needs to be used somewhere in the code, which I don't expect to happen _too_ often.

So I think I'm fine with this approach